Error: Cannot find module 'react-dom/server'

Hello, I am trying to use resend on my server, but I am getting an error that is preventing me from even starting the server. The error seems to be related to react-email, but I don't understand why this would be necessary. I would like to use HTML for sending emails. Could you please help me understand what I need to do to fix this issue? Thank you for your assistance.

> I also tried downgrading the version from 3.5.0 to 3.4.0 and then to 3.2.0, but the error continued to appear

Error:

```
Error: Cannot find module 'react-dom/server'
Require stack:
- C:\Users\xxx\Desktop\app\server\node_modules\@react-email\render\dist\node\index.js
- C:\Users\xxx\Desktop\app\server\node_modules\resend\dist\index.js
- C:\Users\xxx\Desktop\app\server\dist\utils\resend\resend.module.js
- C:\Users\xxx\Desktop\app\server\dist\app.module.js
- C:\Users\xxx\Desktop\app\server\dist\main.js
    at Function.Module._resolveFilename (node:internal/modules/cjs/loader:1144:15)
    at Function.Module._load (node:internal/modules/cjs/loader:985:27)
    at Module.require (node:internal/modules/cjs/loader:1235:19)
    at require (node:internal/modules/helpers:176:18)
    at Object.<anonymous> (C:\Users\xxx\Desktop\app\server\node_modules\@react-email\render\dist\node\index.js:74:30)
    at Module._compile (node:internal/modules/cjs/loader:1376:14)
    at Object.Module._extensions..js (node:internal/modules/cjs/loader:1435:10)
    at Module.load (node:internal/modules/cjs/loader:1207:32)
    at Function.Module._load (node:internal/modules/cjs/loader:1023:12)
    at Module.require (node:internal/modules/cjs/loader:1235:19)
```

Deps:
````
    "dependencies": {
        "@nestjs/common": "^10.0.0",
        "@nestjs/config": "^3.2.1",
        "@nestjs/core": "^10.0.0",
        "@nestjs/event-emitter": "^2.0.4",
        "@nestjs/jwt": "^10.2.0",
        "@nestjs/mongoose": "^10.0.4",
        "@nestjs/passport": "^10.0.3",
        "@nestjs/platform-express": "^10.0.0",
        "@nestjs/platform-socket.io": "^10.3.8",
        "@nestjs/throttler": "^5.1.2",
        "@nestjs/websockets": "^10.3.8",
        "bcrypt": "^5.1.1",
        "cookie-parser": "^1.4.6",
        "mongoose": "^8.2.4",
        "nestjs-zod": "^3.0.0",
        "passport": "^0.7.0",
        "passport-jwt": "^4.0.1",
        "reflect-metadata": "^0.2.0",
        "resend": "^3.5.0",
        "rxjs": "^7.8.1",
        "zod": "^3.22.4"
    },
````
    
I just trying to use this dynamic module:

```@Module({})
export class ResendModule {
    public static forRoot(key: string): DynamicModule {
        const resend = new Resend(key);

        const resendProvider: Provider = {
            provide: 'RESEND_CLIENT',
            useValue: resend,
        };

        return {
            module: ResendModule,
            providers: [resendProvider],
            exports: [resendProvider],
            global: true,
        };
    }
}

--- a/package.json
+++ b/package.json
@@ -1,6 +1,6 @@
 {
   "name": "resend",
-  "version": "4.0.0",
+  "version": "4.0.1-alpha.0",
   "description": "Node.js library for the Resend API",
   "main": "./dist/index.js",
   "module": "./dist/index.mjs",

--- a/package.json
+++ b/package.json
@@ -42,7 +42,7 @@
   },
   "homepage": "https://github.com/resendlabs/resend-node#readme",
   "dependencies": {
-    "@react-email/render": "0.0.17"
+    "@react-email/render": "1.0.1"
   },
   "devDependencies": {
     "@biomejs/biome": "1.9.4",

--- a/src/batch/batch.ts
+++ b/src/batch/batch.ts
@@ -1,4 +1,3 @@
-import { renderAsync } from '@react-email/render';
 import type * as React from 'react';
 import type { Resend } from '../resend';
 import type {
@@ -9,6 +8,7 @@ import type {
 } from './interfaces/create-batch-options.interface';
 
 export class Batch {
+  private renderAsync?: (component: React.ReactElement) => Promise<string>;
   constructor(private readonly resend: Resend) {}
 
   async send(
@@ -24,9 +24,19 @@ export class Batch {
   ): Promise<CreateBatchResponse> {
     for (const email of payload) {
       if (email.react) {
-        email.html = await renderAsync(email.react as React.ReactElement);
-        // biome-ignore lint/performance/noDelete: <explanation>
-        delete email.react;
+        if (!this.renderAsync) {
+          try {
+            const { renderAsync } = await import('@react-email/render');
+            this.renderAsync = renderAsync;
+          } catch (error) {
+            throw new Error(
+              'Failed to render React component. Make sure to install `@react-email/render`',
+            );
+          }
+        }
+
+        email.html = await this.renderAsync(email.react as React.ReactElement);
+        email.react = undefined;
       }
     }
 

--- a/src/emails/emails.ts
+++ b/src/emails/emails.ts
@@ -1,4 +1,3 @@
-import { renderAsync } from '@react-email/render';
 import type * as React from 'react';
 import type { Resend } from '../resend';
 import type {
@@ -22,6 +21,7 @@ import type {
 } from './interfaces/update-email-options.interface';
 
 export class Emails {
+  private renderAsync?: (component: React.ReactElement) => Promise<string>;
   constructor(private readonly resend: Resend) {}
 
   async send(
@@ -36,9 +36,20 @@ export class Emails {
     options: CreateEmailRequestOptions = {},
   ): Promise<CreateEmailResponse> {
     if (payload.react) {
-      payload.html = await renderAsync(payload.react as React.ReactElement);
-      // biome-ignore lint/performance/noDelete: <explanation>
-      delete payload.react;
+      if (!this.renderAsync) {
+        try {
+          const { renderAsync } = await import('@react-email/render');
+          this.renderAsync = renderAsync;
+        } catch (error) {
+          throw new Error(
+            'Failed to render React component. Make sure to install `@react-email/render`',
+          );
+        }
+      }
+
+      payload.html = await this.renderAsync(
+        payload.react as React.ReactElement,
+      );
     }
 
     const data = await this.resend.post<CreateEmailResponseSuccess>(

